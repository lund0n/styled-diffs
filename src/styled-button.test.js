/* eslint-env jest */
import React from "react";
import { mount, render } from "enzyme";
import Button from "./styled-button";
import "jest-styled-components";

test("renders default", () => {
  const component = render(<Button>A Button</Button>);
  expect(component).toMatchSnapshot("default.button");
});

test('ignores custom no-html prop', () => {
  const component = mount(<Button primary>A button</Button>)
  const props = component.find('button').props()
  expect(props).not.toEqual(expect.objectContaining({
    primary: expect.anything()
  }))
})

test("renders with custom prop", () => {
  const component = render(
    <Button primary>
      A button
    </Button>
  );
  expect(component).toMatchSnapshot("custom.button");
});

// These get added to component even though we don't want them.
test('ignores custom html prop', () => {
  const component = mount(<Button content width="100px">A button</Button>)
  const props = component.find('button').props()
  expect(props).not.toEqual(expect.objectContaining({
    content: expect.anything(),
    width: expect.anything()
  }))
})

test('renders with html prop', () => {
  const component = render(
    <Button content width="100px">A button</Button>
  )
  expect(component).toMatchSnapshot("html.button");
})
