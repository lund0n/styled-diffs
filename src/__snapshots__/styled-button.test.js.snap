// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom.button 1`] = `
.c0 {
  display: inline-block;
  border-radius: 3px;
  padding: 0.5rem 0;
  margin: 0.5rem 1rem;
  width: 11rem;
  background: steelblue;
  color: papayawhip;
  border: 2px solid papayawhip;
  background: papayawhip;
  color: steelblue;
}

<button
  class="c0"
>
  A button
</button>
`;

exports[`default.button 1`] = `
.c0 {
  display: inline-block;
  border-radius: 3px;
  padding: 0.5rem 0;
  margin: 0.5rem 1rem;
  width: 11rem;
  background: steelblue;
  color: papayawhip;
  border: 2px solid papayawhip;
}

<button
  class="c0"
>
  A Button
</button>
`;

exports[`html.button 1`] = `
.c0 {
  display: inline-block;
  border-radius: 3px;
  padding: 0.5rem 0;
  margin: 0.5rem 1rem;
  width: 100px;
  background: steelblue;
  color: papayawhip;
  border: 2px solid papayawhip;
}

<button
  class="c0"
  content="true"
  width="100px"
>
  A button
</button>
`;
